.github/workflows/flakeguard.yml

```diff
@@ -307,136 +308,127 @@ jobs:
-        uses: actions/upload-artifact@v4.4.3
-        with:
-          path: all_tests.md
-          name: all-summary.md
-          retention-days: 90
+      
+          # Create test results folder if it doesn't exist
+          mkdir -p ci_test_results
+      
+          # Fix flakeguard binary path
+          PATH=$PATH:$(go env GOPATH)/bin
+          export PATH
+      
+          # Aggregate Flakeguard test results
+          flakeguard aggregate-results \
+            --results-path ./ci_test_results \
+            --output-path ./flakeguard-report \
+            --repo-path "${{ github.workspace }}" \
+            --codeowners-path "${{ github.workspace }}/.github/CODEOWNERS" \
+            --max-pass-ratio "${{ inputs.maxPassRatio }}"
+
+          # Print out the summary file
+          echo -e "\nFlakeguard Summary:"
+          jq . ./flakeguard-report/all-test-summary.json          
+
+          # Read the summary from the generated report
+          summary=$(jq -c '.' ./flakeguard-report/all-test-summary.json)
+          echo "summary=$summary" >> $GITHUB_OUTPUT
 
       - name: Upload All Test Results as Artifact
-        if: ${{ fromJson(steps.set_test_results.outputs.all_tests_count) > 0 }}
+        if: ${{ fromJSON(steps.results.outputs.summary).total_tests > 0 }}
         uses: actions/upload-artifact@v4.4.3
         with:
-          path: all_tests.json
+          path: ./flakeguard-report/all-test-results.json
           name: all-test-results.json
           retention-days: 90
       
-      - name: Upload Failed Tests Summary as Artifact
-        if: ${{ fromJson(steps.set_test_results.outputs.all_tests_count) > 0 }}
-        uses: actions/upload-artifact@v4.4.3
-        with:
-          path: failed_tests.md
-          name: failed-summary.md
-          retention-days: 90
-
       - name: Upload Failed Test Results as Artifact
-        if: ${{ fromJson(steps.set_test_results.outputs.failed_tests_count) > 0 }}
+        if: ${{ fromJSON(steps.results.outputs.summary).failed_runs > 0 }}
         uses: actions/upload-artifact@v4.4.3
         with:
-          path: failed_tests.json
+          path: ./flakeguard-report/failed-test-results.json
           name: failed-test-results.json
-          retention-days: 90        
-
-      - name: Upload Failed Test Logs as Artifact
-        if: ${{ fromJson(steps.set_test_results.outputs.failed_tests_count) > 0 }}
+          retention-days: 90 
+          
+      - name: Upload Failed Test Results With Logs as Artifact
+        if: ${{ fromJSON(steps.results.outputs.summary).failed_runs > 0 }}
         uses: actions/upload-artifact@v4.4.3
         with:
-          path: failed_test_logs.json
-          name: failed-test-logs.json
-          retention-days: 90             
+          path: ./flakeguard-report/failed-test-results-with-logs.json
+          name: failed-test-results-with-logs.json
+          retention-days: 7
```

Comment:
@lukaszcl let's keep those for 90 days so if needed, devs can have access for them longer? It will be a shame to lose those in just 7 days